Framework/src/TaskRunner.cxx

```diff
@@ -421,16 +422,32 @@ void TaskRunner::finishCycle(DataAllocator& outputs)
   }
 }
 
+void TaskRunner::updateMonitoringStats(ProcessingContext& pCtx)
+{
+  mNumberMessagesReceivedInCycle++;
+  for (auto&& input : pCtx.inputs()) {
```

Comment:
this might not take into account multi-part messages. I would use InputRecordWalker instead.

Reply:
yes, absolutely, fixing it.